contracts/Plasma.sol

```diff
@@ -0,0 +1,194 @@
+        operators[_op] = _status;
+        return true;
+    }
+
+    function isOperator() public view returns (bool success) {
+        return operators[msg.sender];
+    }
+
+    // blocks
+    struct Block {
+        uint blockIndex;
+        bytes32 merkleRoot;
+        uint256 timestamp;
+    }
+
+    uint public blockCount;
+    mapping(uint => Block) public childChain;
+
+    event BlockSubmitted(uint indexed _blockIndex, address _operator, bytes32 indexed _merkleRoot);
+
+    function submitBlock(bytes32 _merkleRoot, uint _blockIndex) public onlyOperator {
+        require(blockCount + 1 == _blockIndex);
+
+        Block memory newBlock = Block({
+            blockIndex: _blockIndex,
+            merkleRoot: _merkleRoot,
+            timestamp: block.timestamp
+        });
+
+        blockCount = _blockIndex;
+
+        childChain[blockCount] = newBlock;
+
+        BlockSubmitted(_blockIndex, msg.sender, _merkleRoot);
+    }
+
+    // deposits
+
+    mapping(uint => bytes32) public depositIndex;
+    mapping(bytes32 => uint) public depositBalance;
+    uint public depositCount;
+
+    event Deposit(uint indexed _depositIndex, bytes32 indexed _uid, address _depositor, uint _amount);
+
+    function deposit(address _currency, uint _amount) payable public {
+        require(_amount > 0);
+
+        if (_currency == address(0)) {
+            require(_amount == msg.value);
+        } else {
+            throw;
+        }
+        bytes32 uid = keccak256(msg.sender, _currency, depositCount);
+
+        depositIndex[depositCount] = uid;
+        depositBalance[uid] = _amount;
+
+        Deposit(depositCount, uid, msg.sender, _amount);
+
+        depositCount += 1;
+    }
+
+    // withdrawals
+
+    struct ExitTX {
+        address exitor;
+        uint timestamp;
+    }
+
+    PriorityQueue exitsQueue;
+    mapping(bytes32 => ExitTX) public exits;
+
+    event WithdrawDeposit(address _depositor, uint indexed _depositIndex, bytes32 indexed _uid, uint256 indexed _timestamp, uint _amount);
+    event ExitStarted(uint indexed _priority);
+
+    function withdrawDeposit(uint _depositIndex, address _currency) public {
+        bytes32 uid = depositIndex[_depositIndex];
+        require(uid != 0x0);
+
+        // check if msg.sender is the right one to withdraw
+        require(keccak256(msg.sender, _currency, _depositIndex) == uid);
+
+        ExitTX memory exitTX = ExitTX({
+            exitor: msg.sender,
+            timestamp: block.timestamp + 60
+        });
+
+        addExitToQueue(uid, _depositIndex, exitTX);
+
+        uint amount = depositBalance[uid];
+
+        WithdrawDeposit(msg.sender, _depositIndex, uid, block.timestamp, amount);
+    }
+
+    function addExitToQueue(bytes32 _uid, uint _depositIndex, ExitTX memory exitTX) private {
+        // check if the withdrawal has been already requested
+        require(exits[_uid].timestamp == 0);
+
+        uint priority = exitTX.timestamp << 128 | _depositIndex;
```

Comment:
I'd recommend using [this](https://github.com/kfichter/more-minimal-plasma/blob/master/plasma/contracts/PriorityQueue.sol) version of PriorityQueue.sol to avoid having to do this calculation yourself (it's confusing for reviewers).

Reply:
thanks